### PR TITLE
Add initial code, a test, and community health files

### PR DESCRIPTION
**Pull request by d33bs**
This PR adds initial code, a test, and community health files. I tried to keep the code as true to the original as I could while balancing linting rules I've added as part of this project. Please provide special attention to testing, where I feel we need verification that the output looks as expected with the data.

Other notable changes:

- Added GitHub Actions automated testing through pre-commit and pytest
- Added PR and issue templates
- Suggested the use of private security notices for code of conduct violations (where normally we'd need an email address or similar here for discrete communication).
- The license was updated from `CC-1.0` to `BSD-3-Clause` to acknowledge [Creative Commons guidance on CC license use for software](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+# .python-version
```
Comment (jenna-tomkinson) [inline .gitignore:88]: Is this supposed to be commented out? I would assume it is related to the code comment above so should this be uncommented? Same comments for pipenv and poetry.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+# Cython debug symbols
+cython_debug/
+
+# PyCharm
+#  JetBrains specific template is maintained in a separate JetBrains.gitignore that can
+#  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
+#  and can be added to the global gitignore or merged into this file.  For a more nuclear
+#  option (not recommended) you can uncomment the following to ignore the entire idea folder.
+#.idea/
```
Comment (jenna-tomkinson) [inline .gitignore:160]: Same comment here.

**Review by jenna-tomkinson — APPROVED** [review]
Great PR! Exciting to see the start of this! :tada: It is super cool to see the tests work even on a subset of the CFReT data. Just curious, if we change the name of the software at some point, we would need to go back and change all the links here, correct?

File: .gitignore (hunk 1 above)
Comment (d33bs) [reply .gitignore]: Thanks for this comment. This file comes directly from the GitHub [Python.gitignore](https://github.com/github/gitignore/blob/main/Python.gitignore). [Pyenv](https://github.com/pyenv/pyenv) can be used to set local environments per project, which creates a `.python-version` file. I generally don't use these myself, and wouldn't want to impose a specific Python version on development through this file as we're supporting several versions of Python.

For Poetry, my feeling is that we always want to include a `poetry.lock` file to help navigate dependencies together as a project team. This follows the `.gitignore`'s recommendation as well:

> Similar to Pipfile.lock, it is generally recommended to include poetry.lock in version control.

Comment (d33bs) [reply .gitignore]: I'll remove the commented portions to help clean this file up.

File: .gitignore (hunk 2 above)
Comment (d33bs) [reply .gitignore]: Thanks, I'll remove the commented portions to help clean this file up.

**Review by d33bs — COMMENTED** [review]
Thanks @jenna-tomkinson ! I made some changes to the `.gitignore` based on your comments.

> Just curious, if we change the name of the software at some point, we would need to go back and change all the links here, correct?

Yes, any name change for the project would require link and project title reference updates (implies at least one PR post-name change to update).

**Comment by d33bs** [comment]
Thanks @jenna-tomkinson for the review, merging this in!

**Pull request merged**